src/controllers/UsersController.js

```diff
@@ -0,0 +1,74 @@
+import models from '../db/models';
+import TokenAuthenticate from '../helpers/tokenAuthentication';
+import Response from '../helpers/response';
+const { User, Profile } = models;
+import passwordHash from '../helpers/passwordHash';
+import EmailNotificationAPI from '../helpers/EmailNotificationAPI.js'; 
+
+
+/**
+ * @description Defines the actions to for the users endpoints
+ * @class UserController
+ */
+
+
+ class UserController {
+   static async signUp(req, res){
+       let response;
+      const signUpType = 'local';
+    try{
+    const { firstname, lastname, email, username, password, userType, bio, imgURL } = req.body;
+      const hash = await passwordHash(password, 10);
+
+      const signup = await User
+      .create({
+        email,
+        username,
+        password: hash,
+        userType,
+        signUpType,
+        isVerified: 0
+      });
+      const { id, username: registeredUsername, email: registeredEmail, userType: userSignupType }  = signup.dataValues;
+      await Profile
+      .create({
+        userId: id,
+        firstname,
+        lastname,
+        bio,
+        imgURL,
+      });
+      const userDetails = {id, registeredUsername, registeredEmail, userSignupType };
+      const token = await TokenAuthenticate.generateToken(userDetails, '10hr');
+      const emailBody = `<h1>Verification link</h1><br>
```

Comment:
An E-mail template class will be needed to handle this for easy readability and usage